File: codeabbey/003_min_of_two.rb
#!/usr/bin/env ruby
# encoding: utf-8

x = ARGF.file.read.split("\n").map do |line|
  nums = line.split(' ')
  if nums.size == 2
    nums.map { |e| e.to_i }.min
  else
    nil
  end
end.compact.join ' '
puts x

File: scratchpad/javascript/codeabbey/003-min-of-two.js
#! /usr/bin/env node

var input, output;

input = require('fs')
        .readFileSync('./input.txt', 'utf8')
        .split('\n')
        .slice(1, -1)
        .map(function(e) { 
          return e.split(' ').map(function(num) {
            return parseInt(num, 10);
          });
        });

output = input.map(function(nums) {
  return (nums[0] < nums[1]) ? nums[0] : nums[1];
}).join(' ');

console.log(output);
